solr/core/src/test/org/apache/solr/cloud/TestConfigSetsAPI.java

```diff
@@ -369,13 +370,55 @@ public void testUploadWithScriptUpdateProcessor() throws Exception {
 
   }
 
+  @Test
+  public void testUploadWithLibDirective() throws Exception {
+    // Authorization off
+    unprotectConfigsHandler();
+    final String untrustedSuffix = "-untrusted";
+    uploadConfigSetWithAssertions("with-lib-directive", untrustedSuffix, null, null);
+    // try to create a collection with the uploaded configset
+    Throwable thrown = expectThrows(HttpSolrClient.RemoteSolrException.class, () -> {
+      createCollection("newcollection3", "with-lib-directive" + untrustedSuffix,
+          1, 1, solrCluster.getSolrClient());
+    });
+
+    assertThat(thrown.getMessage(), containsString("Underlying core creation failed"));
+
+    // Authorization on
+    final String trustedSuffix = "-trusted";
+    protectConfigsHandler();
+    uploadConfigSetWithAssertions("with-lib-directive", trustedSuffix, "solr", "SolrRocks");
+    // try to create a collection with the uploaded configset
+    CollectionAdminResponse resp = createCollection("newcollection3", "with-lib-directive" + trustedSuffix,
+        1, 1, solrCluster.getSolrClient());
+    
+    SolrInputDocument doc = sdoc("id", "4055", "subject", "Solr");
+    solrCluster.getSolrClient().add("newcollection3", doc);
+    solrCluster.getSolrClient().commit("newcollection3");
+    assertEquals("4055", solrCluster.getSolrClient().query("newcollection3",
+        params("q", "*:*")).getResults().get(0).get("id"));
+  }
+
   protected SolrZkClient zkClient() {
     ZkStateReader reader = solrCluster.getSolrClient().getZkStateReader();
     if (reader == null)
       solrCluster.getSolrClient().connect();
     return solrCluster.getSolrClient().getZkStateReader().getZkClient();
   }
 
+  private void unprotectConfigsHandler() throws Exception {
+    HttpClient cl = null;
+    try {
+      cl = HttpClientUtil.createClient(null);
+      zkClient().setData("/security.json", "{}".getBytes(UTF_8), true);
+    } finally {
+      if (cl != null) {
+        HttpClientUtil.close(cl);
+      }
+    }
+    Thread.sleep(5000); // TODO: Without a delay, the test fails. Some problem with Authc/Authz framework?
```

Comment:
Ok, obviously get rid of this :)

Reply:
Yes, please. The tests already take an hour.

Reply:
Yeah, that is so irritating indeed. I remember we put that sleep of 5s in for the authentication to kick in. I'll reduce the sleep to 1s and see if it works or not. Please don't mind the potential noise on Jenkins due to this, if it fails.

Reply:
This won't work: some test machines are slower than others, etc. Please don't use wall time in tests. If something needs an async await then re-check the remote state every .5 seconds and expire the test if it doesn't pick up the required change within a reasonable time frame. There has to be a way to check whether the remote status has changed (right)?

Reply:
Okay, I'll dig in to figure out. Maybe there is some way. All the authentication/authorization details are way past my memory.

But anyway, I have limited time today, and lots of issues to get to (before 8.4), please see SOLR-13442). There will be a compromise somewhere or the other unless someone can please help me.

Reply:
Wasn't worth wasting my time today over this. I've ignored the test. Lets move on.
I've also added ref guide changes.

Reply:
Well, we're all trying to help one way or another; lack of time is no excuse to add code that will make other people wonder what the heck is going on later.

I have a strong temptation to ban Thread.sleep with forbidden APIs everywhere, including tests. An unconditional sleep is really worthless in a test -- those jenkins VMs are slow and loaded enough that those tests then proceed too soon and fail with false alarms. And on fast machines they make tests run slow when they could happily proceed. A test should be given just as much attention as the code it tests...

Reply:
> Well, we're all trying to help one way or another; lack of time is no excuse to add code that will make other people wonder what the heck is going on later.
I agree. That's why I removed the new test. By the way, a 5 second sleep was already there in another test.

> I have a strong temptation to ban Thread.sleep with forbidden APIs everywhere
Totally agree. I want to do that myself.

Reply:
> lack of time is no excuse to add code that will make other people wonder what the heck is going on later.

Also, I wasn't trying to be so cruel to other people. I had left a TODO comment to help them understand "what the heck is going on". :-)

Reply:
A TODO doesn't help if all you see is an exception/ failure in jenkins logs.